platform/security/certificate/security-ocsp-checker/src/main/java/org/codice/ddf/security/ocsp/checker/OcspChecker.java

```diff
@@ -123,127 +144,254 @@ Certificate convertToBouncyCastleCert(X509Certificate cert) throws OcspCheckerEx
+  List<OCSPResp> sendOcspRequests(X509Certificate cert, OCSPReq ocspRequest) {
+    Set<String> urlsToCheck = new HashSet<>();
+    if (ocspServerUrls != null) {
+      urlsToCheck.addAll(ocspServerUrls);
+    }
+
+    // try and pull an OCSP server url off of the cert
+    urlsToCheck.addAll(getOcspUrlsFromCert(cert));
+
+    List<OCSPResp> ocspResponses = new ArrayList<>();
+
+    for (String ocspServerUrl : urlsToCheck) {
+      if (isNotBlank(ocspServerUrl)) {
+        try {
+          SecureCxfClientFactory cxfClientFactory =
+              factory.getSecureCxfClientFactory(ocspServerUrl, WebClient.class);
+          WebClient client =
+              cxfClientFactory
+                  .getWebClient()
+                  .accept("application/ocsp-response")
+                  .type("application/ocsp-request");
+
+          Response response = client.post(ocspRequest.getEncoded());
+          ocspResponses.add(createOcspResponse(response));
+          continue;
+        } catch (IOException | OcspCheckerException | ProcessingException ignore) {
+        }
+      }
+      ocspResponses.add(null); // if ocspServerUrl is null or if there was an exception
+    }
+
+    return ocspResponses;
+  }
+
+  /**
+   * Attempts to grab additional OCSP server urls off of the given {@param cert}.
+   *
+   * @param - the {@link X509Certificate} to check.
+   * @return {@link List} of additional OCSP server urls found on the given {@param cert}.
+   */
+  private List<String> getOcspUrlsFromCert(X509Certificate cert) {
+    List<String> ocspUrls = new ArrayList<>();
+
+    try {
+      byte[] authorityInfoAccess = cert.getExtensionValue(Extension.authorityInfoAccess.getId());
+
+      if (authorityInfoAccess == null) {
+        return ocspUrls;
+      }
+
+      AuthorityInformationAccess authorityInformationAccess =
+          AuthorityInformationAccess.getInstance(
+              X509ExtensionUtil.fromExtensionValue(authorityInfoAccess));
+
+      if (authorityInformationAccess == null) {
+        return ocspUrls;
+      }
+
+      for (AccessDescription description : authorityInformationAccess.getAccessDescriptions()) {
+        GeneralName accessLocation = description.getAccessLocation();
+        if (accessLocation.getTagNo() == GeneralName.uniformResourceIdentifier)
+          ocspUrls.add(((DERIA5String) accessLocation.getName()).getString());
+      }
+    } catch (IOException ignore) {
+    }
+
+    return ocspUrls;
+  }
+
+  /**
+   * Creates a {@link OCSPResp} from the given {@param response}.
+   *
+   * @param response - the {@link Response} to convert.
+   * @return an {@link OCSPResp} of the given {@param response}.
+   * @throws OcspCheckerException if any error occurs.
+   */
+  private OCSPResp createOcspResponse(Response response) throws OcspCheckerException {
     Object entity = response.getEntity();
 
     if (!(entity instanceof InputStream)) {
-      postErrorEvent(
```

Comment:
❓ Why did we remove the post to admin notifications in several places? Am I missing something?

Reply:
🐳 Whoops, didn't even notice. Fixing now.